src/backend/mod.rs

```diff
@@ -15,3 +19,68 @@ pub use settings::{Settings, SettingsImpl};
 pub use wallpaper::{Wallpaper, WallpaperImpl, WallpaperOptions};
 
 pub(crate) const IMPL_PATH: &str = "/org/freedesktop/portal/desktop";
+
+// We use option to be able to take() without cloning. Unwraping is safe as they
+// are set in construction.
+pub struct Backend {
+    cnx: Option<zbus::Connection>,
+    name: Option<WellKnownName<'static>>,
+}
+
+impl Backend {
+    pub async fn new<N: TryInto<WellKnownName<'static>>>(name: N) -> Result<Self, crate::Error>
+    where
+        zbus::Error: From<<N as TryInto<WellKnownName<'static>>>::Error>,
+    {
+        Self::new_inner(name).await.map_err(From::from)
```

Comment:
What is the purpose of new_inner? nothing seems to use it ...

Reply:
The purpose is to have something that returns a zbus::Result to use `?`, the tryinto for N is a bit wonky.